tests/buffer/buffer_data_tracking.cpp

```diff
@@ -0,0 +1,160 @@
+/* RUN: %{execute}%s | %{filecheck} %s
+   CHECK: Result:
```

Comment:
Since there is no `Result:` in the output, this line is not required and makes the test to fail.
I realize that you do not run the tests with `make`.
Look at the `tests/README.rst`. Basically you just need to do in the `tests` directory:
```
make clean check
```